src/main/java/io/anserini/search/SearchDenseVectors.java

```diff
@@ -341,22 +344,43 @@ public <K> void runTopics() throws IOException {
 
 
     LOG.info("============ Launching Search Threads ============");
-
+    ThreadPoolExecutor searcherExecutor;
+    if (args.threadsPerQuery > 1) {
+      searcherExecutor = (ThreadPoolExecutor) Executors.newFixedThreadPool(args.threadsPerQuery);
+    } else {
+      searcherExecutor = null;
```

Comment:
if you set `ThreadPoolExecutor searcherExecutor = null` above, you won't need this branch?